sound/soc/sof/intel/hda-loader.c

```diff
@@ -323,6 +323,9 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 	struct firmware stripped_firmware;
 	int ret, tag;
 
+	/* disable clock gating and power gating before fw boot */
+	hda_dsp_ctrl_clock_power_gating(sdev, false);
+
```

Comment:
is this needed before fw boot or before fw download?
In the first case you could move this to the run case...

Reply:
@plbossart this is needed before fw boot. 

Could you please explain what you mean by move this to the run case? For apl, the run ops is set to hda_dsp_cl_boot_firmware(). So thats why I added it in here.

Reply:
If this is needed before fw_boot, then it should be moved before the fw_boot and not before the code loader.

Reply:
@plbossart sorry I was wrong earlier. This needs to be disabled before fw download and re-enabled after fw boot. I will move it to the run routine with pre_load and post_lost ops.